src/size.c

```diff
@@ -98,6 +98,30 @@ SEXP vctrs_size(SEXP x) {
   return Rf_ScalarInteger(vec_size(x));
 }
 
+SEXP list_sizes(SEXP x) {
+  if (!vec_is_list(x)) {
+    Rf_errorcall(R_NilValue, "`x` must be a list.");
+  }
+
+  R_len_t size = vec_size(x);
+
+  SEXP out = PROTECT(Rf_allocVector(INTSXP, size));
+  int* p_out = INTEGER(out);
+
+  for (R_len_t i = 0; i < size; ++i) {
+    SEXP elt = VECTOR_ELT(x, i);
+    p_out[i] = vec_size(elt);
+  }
+
+  UNPROTECT(1);
+  return out;
+}
+
+// [[ register() ]]
+SEXP vctrs_list_sizes(SEXP x) {
```

Comment:
Minor: Do we need a registered alias since they have the same signature?

Reply:
I tend to try and always keep a separate function prefixed with `vctrs_` that gets exported, whether it is strictly required or not